lib/std/array_list.zig

```diff
@@ -355,12 +376,22 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
     };
 }
 
-/// Bring-your-own allocator with every function call.
-/// Initialize directly and deinitialize with `deinit` or use `toOwnedSlice`.
+/// An `ArrayList`, but with bring-your-own allocator on every function call.
```

Comment:
```suggestion
/// An `ArrayList`, but the allocator is passed as a parameter to relevant functions rather than
/// stored in the struct itself. The same allocator must be used throughout the entire lifetime of
/// an ArrayList.
```